packages/flutter_tools/lib/src/commands/config.dart

```diff
@@ -20,6 +22,11 @@ class ConfigCommand extends FlutterCommand {
       help: 'Clear the saved development certificate choice used to sign apps for iOS device deployment.');
     argParser.addOption('android-sdk', help: 'The Android SDK directory.');
     argParser.addOption('android-studio-dir', help: 'The Android Studio install directory. If unset, flutter will search for valid installs at well-known locations.');
+    argParser.addOption('jdk-dir', help: 'The Java Development Kit (JDK) install/home directory. '
```

Comment:
The `-dir` on the end of the name seems a bit redundant/unnecessary, but it matches convention the prexisting `android-studio-dir` and `build-dir` option (which `android-sdk` annoyingly does not follow).

I suppose we could always unify these and then supply the old names in the `aliases` argument to maintain backwards compatibility. I digress.

Reply:
unifying while not breaking by using aliases SGTM

Reply:
I think you can remove the trailing space.

Also, `"install/home directory"` seems needlessly ambiguous. Can we just say `"install directory."`?

Reply:
> I think you can remove the trailing space.

The trailing space is necessary to satisfy [missing_whitespace_between_adjacent_strings](https://dart-lang.github.io/linter/lints/missing_whitespace_between_adjacent_strings.html)

> Also, `"install/home directory"` seems needlessly ambiguous.

Agreed. Updated to 'installation directory'.

Reply:
> The trailing space is necessary to satisfy [missing_whitespace_between_adjacent_strings](https://dart-lang.github.io/linter/lints/missing_whitespace_between_adjacent_strings.html)

Whoops, good thing it was there too, totally misread this.